code/datums/uplink/uplink_sources.dm

```diff
@@ -58,21 +58,11 @@ var/global/list/default_uplink_source_priority = list(
 	if(!R)
 		return SETUP_FAILED
 
-	var/freq = PUBLIC_LOW_FREQ
-	var/list/freqlist = list()
-	while (freq <= PUBLIC_HIGH_FREQ)
-		if (freq < 1451 || freq > PUB_FREQ)
-			freqlist += freq
-		freq += 2
-		if ((freq % 2) == 0)
-			freq += 1
-
-	freq = freqlist[rand(1, freqlist.len)]
 	var/obj/item/uplink/T = new(R, M.mind, amount)
 	R.hidden_uplink = T
-	R.traitor_frequency = freq
-	to_chat(M, "<span class='notice'>A portable object teleportation relay has been installed in your [R.name]. Simply dial the frequency [format_frequency(freq)] to unlock its hidden features.</span>")
-	M.StoreMemory("<B>Radio Freq:</B> [format_frequency(freq)] ([R.name]).", /decl/memory_options/system)
+	R.traitor_frequency = rand(PUBLIC_LOW_FREQ+1, PUB_FREQ-1)
```

Comment:
Uh, this will result in invalid frequencies, won't it? You can only select odd numbers, but this can select even ones.

Reply:
This PR removed that restriction.